File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-1.kt
class Solution {
    fun characterReplacement(s: String, k: Int): Int {

        var left = 0
        var result = 0
        var maxFreq = 0

        val map = IntArray(26)

        for (right in s.indices) {

            val idx = s[right] - 'A'
            map[idx]++
            maxFreq = maxOf(maxFreq, map[idx])

            while (right - left + 1 - maxFreq > k) {
                map[s[left] - 'A']--
                left++
            }

            result = maxOf(result, right - left + 1)
        }

        return result
    }
}

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-1.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        
        result = [1] * len(nums)

        left = 1
        for i in range(len(nums)):
            result[i] = left
            left *= nums[i]

        right = 1
        for i in reversed(range(len(nums))):
            result[i] *= right
            right *= nums[i]

        return result